### DISCUSSION TITLE
Problem - Self-Use Mode always automatically toggles back to on

### DISCUSSION THREAD

**Discussion by japers** [Pho3niX90/solis_modbus#70] [Q&A]
Thanks Pho3niX90 for a great integration!

I have a very off problem since moving over from the Solax integration:

Within 60 seconds of toggling "Self Use Mode" off (as per below), it toggles back to on (not caused by any other automation). The result being that the battery stops charging.

<img width="358" alt="image" src="https://github.com/user-attachments/assets/c8f28691-82d2-436c-a9d2-85d3ff995b34">

This was not an issue with the Solax integration. I therefore think I must be missing something?

**Comment by Pho3niX90** [comment]
The toggles are most of the time interlinked by other settings, which cannot be properly visualized by buttons alone. 

For example, to allow button X to be on, button Y need to be off. Otherwise the Solis will not accept the state change. 
The second issue, is not all solis make use of the same registrars, and on most is just reserve. 

Which Solis are you using?

**Reply by morts1** [reply]
Hi, thank you so much for such a great integration! I'm currently trying to understand the proper order for changing between different inverter modes as you say they are interlinked. 
From what you state above, I assume after turning off self-use, another mode needs to be enabled within a time period, otherwise it reverts back to self-use?
